### ISSUE TITLE
CH552: Unknown USB after downloading firmware

### ISSUE THREAD

**Issue by Bruur-vr** [DeqingSun/ch55xduino#172]
Hello,

I bought a macro pad board with a CH552G i'd like to reflash. I did not like the firmware it came with (I had no way of changing buttons) so I tried overwriting it through ArduinoIDE with ch55xduino. I installed everything in ArduinoIDE, selected board and adjusted parameters under "tools". Then hit upload while it was in bootloader mode.  The uploading went succesfully.

Ever since it shows as Unknown USB device.

![afbeelding](https://github.com/user-attachments/assets/1f00a2d4-e16f-4aec-88ec-22247b0db314)

"Windows has stopped this device because it has reported problems. (Code 43)"

Problem code is 0000002B

Events says this:
```
Device USB\VID_0000&PID_0005\5&38e97a59&0&7 was not migrated due to partial or ambiguous match.

Last Device Instance Id: USB\VID_0000&PID_0003\6&31b1809d&0&1
Class Guid: {36fc9e60-c465-11cf-8056-444553540000}
Location Path: 
Migration Rank: 0xF000FFFFFFFFF100
Present: false
Status: 0xC0000719
```

in usbview it shows up like this:
![afbeelding](https://github.com/user-attachments/assets/a8f6b98a-5e8e-4557-b6a1-8c8e3af53f94)

Zadig shows up as it having no driver when out of bootloader mode.
![afbeelding](https://github.com/user-attachments/assets/87fe75d5-04f1-4e07-930c-72a919a0e8c5)

When in bootloader it shows this, after I tried reïnstalling drivers (WINusb & libusb) but it makes no difference.
![afbeelding](https://github.com/user-attachments/assets/3e8dbfc6-5fc6-45f9-bff3-86e8ba425ba5)

I tried Tools > "export compiled binary" in ArduinoIDE, and it gives a hex file.
I also tried putting this in WCHISPstudio but after flashing I get the same "USB device not recognized" message.
![afbeelding](https://github.com/user-attachments/assets/16d0ea96-244a-41ec-8c9e-0586a2c70a8f)

The only way I get it to be recognized is by flashing it through CMD with the line:
`python ./tools/chprog.py macropad_plus.bin`
When using the makefile & .bin supplied by this github [page](https://github.com/wagiminator/CH552-MacroPad-plus).
The problem is, the macro's assigned to the buttons need to be changed in the sketch, and reuploaded. And when I try reuploading with adjustments it gives unknown USB device.

I tried changing the macropage_plus.bin to the Compiled file ArduinoIDE gives me, but that makes the CH552 undetectable in either bootloader or not unless I reinstall CH372 driver.

I am not well versed in these kind of topics so forgive me for not understanding in the first try.

TLDR; When uploading through ArduinoIDE I get unknown USB. I tried other tools but the result remains the same.

EDIT: reopened after editing first post, since it is most likely a flash problem.

**Comment by Bruur-vr** [comment]
I am sorry, I posted this to the wrong github page.

**Comment by DeqingSun** [comment]
It seems:
You can enter bootloader mode, and flash a firmware.
You can flash a stock firmware from the vendor, it works.
So there seems to be no hardware problem. The most possible reason is you compiled your firmware to run at 24MHz, while your voltage for CH552 is only 3.3V, or you have a bad capacitor. 
The CH552-MacroPad-plus compiled the firmware for 16MHz. 
Confirm the voltage of VCC on CH552, then use drop down menu to choose 16M speed if the voltage is 3.3V.

**Comment by Bruur-vr** [comment]
Hello,
The VCC at the bootswitch is 3.17V
I turned on "16MHz internal" before uploading.

**Comment by DeqingSun** [comment]
The example does not work on my setup either, I got the same error. 
Let me check it.

**Comment by DeqingSun** [comment]
Downgrade your Arduino to 0.0.16, which is the version that [wagiminator](https://github.com/wagiminator) prepared the code. It worked for me.

**Comment by Bruur-vr** [comment]
> Downgrade your Arduino to 0.0.16, which is the version that [wagiminator](https://github.com/wagiminator) prepared the code. It worked for me.

Thanks for the advice, I downloaded the 0016 IDE, this IDE has no board manager where do I add the CH55x board?
Also: I cannot upload the sketch that was included for the macropad in this version of the IDE. I doubt it was originally written for this version?

**Comment by DeqingSun** [comment]
Sorry I mean downgrade CH55xduino to 0.0.16

**Comment by Bruur-vr** [comment]
> Sorry I mean downgrade CH55xduino to 0.0.16

This seemed to fix it. Thank you very much.
